fontes/avaliador-sintatico/avaliador-sintatico.ts

```diff
@@ -1285,7 +1300,9 @@ export class AvaliadorSintatico implements AvaliadorSintaticoInterface<SimboloIn
                 return this.funcao('funcao');
             }
 
-            if (this.verificarSeSimboloAtualEIgualA(tiposDeSimbolos.CLASSE)) return this.declaracaoDeClasse();
+            if ([tiposDeSimbolos.CLASSE, tiposDeSimbolos.ARROBA].includes(this.simboloAtual().tipo)) {
```

Comment:
E se for função? 

Aquilo que te falei antes: eu acho que seria melhor ter um `resolverDecorador` quando uma arroba é encontrada, empilhando isso numa variável interna, por exemplo, `pilhaDecoradores`. Aí quando uma classe ou função é encontrada, você desempilha a pilha toda pra dentro do construtor da declaração de classe/função.